backend/fief/services/email/generic.py

```diff
@@ -0,0 +1,44 @@
+import smtplib, ssl
+from email.message import EmailMessage
+from typing import Optional, Tuple
+from fief.services.email.base import EmailProvider, SendEmailError
+
+
+class Generic(EmailProvider):
+    def __init__(
+        self, host: str, username: str, password: str, port: int = 587
+    ) -> None:
+        self.username = username
+        self.password = password
+        self.host = host
+        self.port = port
+
+    def send_email(
+        self,
+        *,
+        sender: Tuple[str, Optional[str]],
+        recipient: Tuple[str, Optional[str]],
+        subject: str,
+        html: Optional[str] = None,
+        text: Optional[str] = None,
+    ):
+        from_email, from_name = sender
+        to_email, to_name = recipient
+
+        try:
+            message = EmailMessage()
+            message["Subject"] = subject
+            message["From"] = self.format_address(from_email, from_name)
+            message["To"] = self.format_address(to_email, to_name)
+            message.set_content(text)
+            message.add_alternative(html, subtype="html")
+
+            # TODO: Cope with multiple setups? Not just TLS?
```

Comment:
I guess that's the most common way nowadays. Maybe we could be a bit more flexible by making `username`, `password` optional and have an `ssl` flag? Something like this:

```py
def __init__(
        self, host: str, username: Optional[str] = None, password: Optional[str] = None, port: int = 587
    ) -> None:
```

```py
with smtplib.SMTP(self.host, self.port) as server:
    if self.ssl:
        server.starttls(context=context)
    if self.username and self.password:
        server.login(self.username, self.password)
    server.send_message(message)
```

Reply:
Nice idea :)